src/lib/device-types/index.ts

```diff
@@ -186,6 +204,12 @@ function fetchDeviceTypesAndReschedule(): Promise<Dictionary<DeviceTypeInfo>> {
 	// in case another api request comes before the first completes
 	if (!deviceTypesCache) {
 		deviceTypesCache = promise;
+		// make sure we remove rejected promises from the cache
+		promise.catch(() => {
+			if (deviceTypesCache === promise) {
+				deviceTypesCache = undefined;
```

Comment:
The initial request is only one that is assigned to the cache before it resolves.
Before this PR we would retry fetching the device types forever, so the cached promise could never settle as rejected.
We now have to handle make sure to clear the cache if the initial request fails.

Reply:
Any reason for not using memoizee promise mode for caching the device types?  It will automatically handle clearing the cache on failure

Reply:
@Page-  The only reason I did this manually since the first iteration is that memoizee doesn't provide a "cache-then-update" caching schema.
The reasoning is that after the API boots and gets a warm cache, we prefer to have near zero cost on accessing the deviceTypes, since it's used in various endpoints as well as in hooks.
If we simply went the `memoizee` road, even when using `prefetch` we could end up in situations that incoming requests are waiting as long as 5 secs for the cache to be repopulated.

For other not so critical part, like fetching all the available versions of a device type, we do use memoizee on the s3 requests.

The img maker is doing something similar as well, so I guess the reasoning might be similar.